src/test/scala/mesosphere/marathon/api/v2/json/AppDefinitionFormatsTest.scala

```diff
@@ -411,15 +408,7 @@ class AppDefinitionFormatsTest extends UnitTest
           |    }
           |  }
           |}""".stripMargin).as[raml.App]
-      ) should failWith(
-        GroupViolationMatcher(
-          description = "container",
-          constraint = "is invalid",
-          violations = Set(
-            group("docker", "is invalid", "credential" -> "must be empty")
-          )
-        )
-      )
+      shouldViolate(app, "/container/docker/credential" -> "must be empty")(AppValidation.validateOldAppAPI)
```

Comment:
Nit: As a follow up we could make it a ScalaTest assertion:
```scala
AppValidation.validateOldAppAPI shouldViolate("/container/docker/credential" -> "must be empty") on(app)
```

or something similar. I would not do so in this diff though.

Reply:
I agree; I'd like to do it in this diff.